lib/internal/errors.js

```diff
@@ -610,6 +610,17 @@ function addNumericalSeparator(val) {
   return `${val.slice(0, i)}${res}`;
 }
 
+// Append `causedByError` to the `error`.
+function addCausedByError(error, message, causedByError) {
+  let msg = message;
+  if (causedByError) {
```

Comment:
It might actually make sense to remove this `if` statement. If the cause is e.g., a thrown `undefined` or `false`, so be it.